lang/java/avro/src/main/java/org/apache/avro/util/Utf8.java

```diff
@@ -119,16 +120,21 @@ public Utf8 setByteLength(int newLength) {
     }
     this.length = newLength;
     this.string = null;
-    this.hasHash = false;
+    this.hash = 0;
     return this;
   }
 
   /** Set to the contents of a String. */
   public Utf8 set(String string) {
-    this.bytes = getBytesFor(string);
-    this.length = bytes.length;
+    byte[] bytes = getBytesFor(string);
+    int length = bytes.length;
+    if (length > MAX_LENGTH) {
+      throw new AvroRuntimeException("String length " + length + " exceeds maximum allowed");
+    }
```

Comment:
Im not sure, but this seemed like a bug to me as well. 
`MAX_LENGTH` should be kept at all time right?

Reply:
If it is, the other 2 constructors need to be changed as well.

Reply:
I agree, that seems reasonable to me!  Do you want to make the change in the PR directly or create a JIRA?

Reply:
I just added to this PR :smile: